feat(platform): add copy to clipboard button in workspace overview page

omlahore opened the pull request:
### **User description**
## Description

Added a "Copy to Clipboard" button in the "Store private key with us?" section inside the Workspace Overview page.

Fixes #905

## Dependencies

No new dependencies/packages were added.

## Future Improvements

Could add a toast notification like "Copied!" after the copy action for better UX.

## Mentions

@rajdip-b (issue author)  
@Allan2000-Git (initial assignee)

## Screenshots of relevant screens

_No UI screenshot added as functionality was added in code. Copy button will be visible next to private key once private key is available._

## Developer's checklist

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-check on my work

**If changes are made in the code:**

- [x] I have followed the [coding guidelines](https://google.github.io/styleguide/jsguide.html)
- [x] My changes in code generate no new warnings
- [x] My changes are not breaking another fix/feature of the project
- [ ] I have added test cases to show that my feature works (no test framework visible for frontend)
- [x] I have added relevant screenshots in my PR (explained in text)
- [x] There are no UI/UX issues

### Documentation Update

- [x] This PR does not require documentation updates


___

### **PR Type**
Enhancement


___

### **Description**
- Added "Copy to Clipboard" button for project private key.

- Integrated `CopyToClipboard` component in workspace overview.

- Displayed copy button only when private key is available.

- Improved UI for easier private key copying.


___



### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>page.tsx</strong><dd><code>Add copy-to-clipboard functionality for project private key</code></dd></summary>
<hr>

apps/platform/src/app/(main)/(project)/[workspace]/[project]/@overview/page.tsx

<li>Imported and integrated <code>CopyToClipboard</code> component.<br> <li> Added conditional rendering for copy button when private key exists.<br> <li> Enhanced UI to include "Copy private key" label next to button.<br> <li> Refactored section for improved clarity and usability.


</details>


  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/909/files#diff-c0b170943d4abee3d4aed9ebcaec39b9ee76dd2684c5706a252bd4ae809da27e">+30/-19</a>&nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about Qodo Merge usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

rajdip-b commented on apps/platform/src/app/(main)/(project)/[workspace]/[project]/@overview/page.tsx:
```diff
@@ -214,25 +216,34 @@ function OverviewPage(): React.JSX.Element {
+
+ 
+  {projectPrivateKey ? <div className="flex items-center gap-2">
+      <CopyToClipboard text={projectPrivateKey} />
```
There's a `hidden-component.tsx` file that you should use. As was mentioned in the issue thread

rajdip-b reviewed (changes requested):
"Set up private key in memory" also needs to have the same

omlahore commented:
Updated 'Store private key with us?' and 'Set up private key in memory?' sections to use HiddenContent. Also cleaned unused code. Ready for re-review.

rajdip-b commented:
Code looks good. Could you please post a few screenshots or a single video demonstrating the changes?

rajdip-b commented:
@omlahore hey man, could you please share a screenshare of the feature?

rajdip-b commented:
Closing due to inactivity
